src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateStore.java

```diff
@@ -148,44 +148,46 @@ public X509Certificate getCACertificate() throws KeyStoreException {
         return certChain[0];
     }
 
-    private void setCaCertificateChain(Certificate... caCertificateChain) throws KeyStoreException {
-        if (caCertificateChain == null) {
-            throw new KeyStoreException("No certificate chain provided");
-        }
-        for (Certificate cert : caCertificateChain) {
-            if (!(cert instanceof X509Certificate)) {
-                throw new KeyStoreException("Unsupported certificate type");
-            }
-        }
-        setCaCertificateChain(Arrays.stream(caCertificateChain).toArray(X509Certificate[]::new));
-    }
 
     /**
-     * Sets the CA certificate chain.
+     * Sets the CA chain and private key that are used to generate certificates. It combines setting both values
+     * at the same time to avoid invalid states where the caChain can be updated without updating the value of
+     * the private key required to sign generated certificates.
+     *
+     * @param privateKey  leaf CA private key
+     * @param caCertificateChain a CA chain
      *
-     * @param caCertificateChain Array of CA certificates
-     * @throws KeyStoreException if unable to retrieve the certificate chain
+     * @throws KeyStoreException  if privateKey is not instance of PrivateKey or no ca chain provided
      */
-    public void setCaCertificateChain(X509Certificate... caCertificateChain) throws KeyStoreException {
+    public void setCaCertificateChain(Key privateKey, X509Certificate... caCertificateChain) throws KeyStoreException {
```

Comment:
Lump together the update of the CA chain and private key. If we don't do this we leave the door open for future bugs where we update we have an old private key stores, we call `setCaCertificateChain` which emits events that might end up creating certificates reading the stale value of the key.

A CA should always be accompanied with its private key and change atomically (all at once) to avoid issues.